src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseFVMKernels.cpp

```diff
@@ -663,6 +663,356 @@ INST_FluxKernel( 5, FaceElementStencil );
+                                                          phaseVisc,
+                                                          phaseDens,
+                                                          phaseMassDens,
+                                                          phaseCompFrac,
+                                                          phaseOutflux,
+                                                          compOutflux );
+  } );
+}
+
+#define INST_CFLFluxKernel( NC, STENCIL_TYPE ) \
+  template \
+  void CFLFluxKernel:: \
+    launch< NC, STENCIL_TYPE >( localIndex const numPhases, \
+                                real64 const & dt, \
+                                STENCIL_TYPE const & stencil, \
+                                ElementViewConst< arrayView1d< real64 const > > const & pres, \
+                                ElementViewConst< arrayView1d< real64 const > > const & gravCoef, \
+                                ElementViewConst< arrayView3d< real64 const > > const & phaseRelPerm, \
+                                ElementViewConst< arrayView3d< real64 const > > const & phaseVisc, \
+                                ElementViewConst< arrayView3d< real64 const > > const & phaseDens, \
+                                ElementViewConst< arrayView3d< real64 const > > const & phaseMassDens, \
+                                ElementViewConst< arrayView4d< real64 const > > const & phaseCompFrac, \
+                                ElementView< arrayView2d< real64 > > const & phaseOutflux, \
+                                ElementView< arrayView2d< real64 > > const & compOutflux )
+
+INST_CFLFluxKernel( 1, CellElementStencilTPFA );
+INST_CFLFluxKernel( 2, CellElementStencilTPFA );
+INST_CFLFluxKernel( 3, CellElementStencilTPFA );
+INST_CFLFluxKernel( 4, CellElementStencilTPFA );
+INST_CFLFluxKernel( 5, CellElementStencilTPFA );
+
+INST_CFLFluxKernel( 1, FaceElementStencil );
+INST_CFLFluxKernel( 2, FaceElementStencil );
+INST_CFLFluxKernel( 3, FaceElementStencil );
+INST_CFLFluxKernel( 4, FaceElementStencil );
+INST_CFLFluxKernel( 5, FaceElementStencil );
+
+#undef INST_CFLFluxKernel
+
+/******************************** CFLKernel ********************************/
+
+template< localIndex NP >
+GEOSX_HOST_DEVICE
+GEOSX_FORCE_INLINE
+void
+CFLKernel::
+  computePhaseCFL( real64 const & poreVol,
+                   arraySlice1d< real64 const > phaseRelPerm,
+                   arraySlice2d< real64 const > dPhaseRelPerm_dPhaseVolFrac,
+                   arraySlice1d< real64 const > phaseVisc,
+                   arraySlice1d< real64 const > phaseOutflux,
+                   real64 & phaseCFLNumber )
+{
+  // first, check which phases are mobile in the cell
+  real64 mob[NP]{};
+  localIndex mobilePhases[NP]{};
+  localIndex numMobilePhases = 0;
+  for( localIndex ip = 0; ip < NP; ++ip )
+  {
+    mob[ip] = phaseRelPerm[ip] / phaseVisc[ip];
+    if( mob[ip] > 1e-12 )
```

Comment:
Maybe make it a class-level constant? It's not really used anywhere else, just tells the reader we consider this value somewhat important.
```c++
static constexpr real64 minPhaseMobility = 1e-12;
```

Reply:
I added this class-level constant to check the zero mobilities.